### ISSUE TITLE
Inference smoothing creates border effects

### ISSUE THREAD

**Issue by remtav** [NRCan/geo-deep-learning#299]
At inference, our [smoothing window](https://github.com/NRCan/geo-deep-learning/blob/develop/inference_segmentation.py#L181) multiplies raw chip probability maps to set higher confidence to predictions in center of chip. As a result of the convolution with a high overlap (low stride) and smoothing on original image, it seems the smoothing may overestimate class probabilities at upper-right corner as is shown by this heatmap on which a water extraction never "activates" (i.e. predicts with constant probability the absence of waterbody). This should be tested thoroughly and independently of the action of any model (i.e. with a numpy array filled with close-to-zero values).
![image](https://user-images.githubusercontent.com/34774759/160870096-091e5358-57d5-4eb2-aba9-6f8c28e1f67f.png)

**Comment by remtav** [comment]
This issue is once more confirmed:
![image](https://user-images.githubusercontent.com/34774759/204630265-de5afc7f-0c40-4c50-b938-23f8002428bb.png)

As can be seen here on the heatmap from a "Forest" extraction on a WorldView image in Nova Scotia (id=NS6), the upper-right corner of heatmap has higher confidence values than the rest of extraction.

Another symptom than can be observed by this border effect is a fading towards lower values of confidence on the 4 borders of extractions. Screenshots below demonstrate this: the heatmap extend to the border of valid data in imagery, but after threshold, because of that "fading out" effect, a portion of the border is cropped although the extraction was valid at that place. A much larger padding should solve this.

Imagery:
![image](https://user-images.githubusercontent.com/34774759/204649131-a060731b-1856-4929-9dc5-c02bcef19bd5.png)

Heatmap (from raw logits, after softmax):
![image](https://user-images.githubusercontent.com/34774759/204649219-a8157e5e-db0f-4ef6-8613-69128587b54d.png)

Inference from thresholded heatmap (overlayed on top of heatmap, showing the "cropped" border from raw extraction):
![image](https://user-images.githubusercontent.com/34774759/204649269-3264d204-822e-4813-bcbf-dcc9f238abc1.png)

**Comment by remtav** [comment]
closed with #439
